File: from-2501-to-2600/2554-maximum-number-of-integers-to-choose-from-a-range-i/solution.go
func maxCount(banned []int, n int, maxSum int) int {
    isBanned := make(map[int]bool)
    for _, b := range banned {
        if b <= n {
            isBanned[b] = true
        }
    }
    sum, cnt := 0, 0
    for i := 1; i <= n; i ++ {
        if !isBanned[i] && i + sum <= maxSum {
            cnt += 1
            sum += i
        }
    }
    return cnt
}

File: from-2501-to-2600/2554-maximum-number-of-integers-to-choose-from-a-range-i/solution.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        isBanned = defaultdict(bool)
        for b in banned:
            if b <= n:
                isBanned[b] = True

        sum, cnt = 0, 0
        for i in range(1, n + 1):
            if not isBanned[i] and i + sum <= maxSum:
                cnt += 1
                sum += i

        return cnt